content/developer/tutorials/server_framework_101/06_security.rst

```diff
@@ -348,10 +348,128 @@ access to property records.
 Separate company data
 =====================
 
-tmp
+In enterprise environments, organizations often need to manage multiple distinct entities within the
+same system. This approach allows different branches, subsidiaries, franchises, or even completely
+different companies to operate independently while sharing common resources and functionalities.
+
+In Odoo, the **multi-company** feature enables managing multiple companies within a single database.
+Each company can have its own configuration and data, while still allowing users to access data from
+multiple companies. This is implemented through several key mechanisms:
+
+- **Company field**: Adding a `company_id` many-to-one field to a model allows linking its records
+  to a specific company (represented by the generic `res.company` model). Records can be:
+
+  - **Company-specific**: When `company_id` has a value, making the record belong to one company.
+  - **Company-shared**: When `company_id` is empty, making the record accessible across all
+    companies.
+
+- **Company-dependent fields**: The `company_dependent=True` attribute set on a field creates a
+  separate value for each company. The values are stored in a JSON object in the database and the
+  right value is automatically selected based on the current company of the user.
+- **Company context**: The `with_company(company)` model method temporarily changes the company
+  context when accessing company-dependent fields, allowing to retrieve values from a specific
+  company's perspective.
+- **Context-aware dependencies**: The `@api.depends_context('company')` decorator ensures that
+  computed fields are computed depending on the current company (`self.env.company`).
```

Comment:
are cached (and computed) depending on the environment company.
Maybe a tip to say that it shouldn't be used on stored fields (unless you really know what you're doing)

Reply:
I prefer to omit it for now because we're in the scope of a tutorial where the reader has to learn pretty much everything. It should however be properly state in the reference doc.